docs/cql/ddl.rst

```diff
@@ -978,6 +978,19 @@ The ``ALTER TABLE`` statement can:
 .. warning:: Once a column is dropped, it is allowed to re-add a column with the same name as the dropped one
    **unless** the type of the dropped column was a (non-frozen) column (due to an internal technical limitation).
 
+It is also possible to drop a column with specified timestamp ``ALTER TABLE ... DROP ... USING TIMESTAMP ...``.
+The purpose of this statement is to be able to safely restore schema in the case a column was dropped and re-added later.
+The timestamp should be obtained by describing schema with internals ``DESC SCHEMA WITH INTERNALS`` (or other descriptions like ``DESC TABLE ks.cf WITH INTERNALS``)
+
+For example: 
+Let's say you have a table with some data. Then you drop one of the column and re-add it later.
+In the future, when you wish to restore the schema, you **have to** also drop the column with specified timestamp (the same timestamp as the original drop)
+and re-add it again.
+Otherwise, you can ressurect your data (if you skip ``ALTER ... DROP/ADD ...`` entirely) 
+or you can lost data inserted after column re-addition (if you drop the column without the timestamp).
+
+.. warning:: Dropping a column with specified timestamp should only be used to restore schema from description (``DESCRIBE SCHEMA WITH INTERNALS``).
```

Comment:
I think it would make sense to put a link to our documentation about restoring schema.